tasks/main.yml

```diff
@@ -1,3 +1,209 @@
+  user:
+    name: "{{ devspace_user }}"
+    comment: "Devspace maintainer"
+    uid: "{{ devspace_user_id }}"
+    groups: docker
+    append: yes
+
+- name: ensures /home/{{ devspace_user }}/.ssh dir exists
+  become: yes
+  become_user: "{{ devspace_user }}"
+  file:
+    path: "/home/{{ devspace_user }}/.ssh"
+    state: "directory"
+    owner: "{{ devspace_user }}"
+    group: "{{ devspace_user }}"
+    mode: 0700
+
+- name: allow {{ devspace_user }} to ssh directly
+  become: yes
+  copy:
+    remote_src: yes
+    src: "{{ ansible_env.HOME }}/.ssh/authorized_keys"
+    dest: "/home/{{ devspace_user }}/.ssh/authorized_keys"
+    owner: "{{ devspace_user }}"
+    group: "{{ devspace_user }}"
+
+- name: install python tools
+  become: yes
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - python-devel
+    - python-pip
+
+- name: install python packages
+  become: yes
+  pip:
+    name: "{{ item }}"
+    #extra_args: "--upgrade"
+    state: latest
+  with_items:
+    - pip
+    - passlib
+    - pyyaml
+    - docker-compose==1.9.0
+
+- name: clone devspace
+  become: yes
+  become_user: "{{ devspace_user }}"
+  git:
+    repo: "{{ devspace_git_repo }}"
+    dest: "{{ devspace_home }}"
+    update: "{{ devspace_git_update }}"
+    force: "{{ devspace_git_force }}"
+    refspec: "+refs/pull/*:refs/heads/*"
+    version: "{{ devspace_git_repo_version }}"
+  register: gitstatus
+
+
+# TASK [openmicroscopy.ci-devspace : Devspace stop] **********************************************************************************
+# fatal: [a20ea3a7-7d96-4db1-aed3-c4f9de103313]: FAILED! => {"changed": false, "failed": true, "msg": "Error connecting: Error while fetching server API version: ('Connection aborted.', error(13, 'Permission denied'))"}
+- name: Devspace stop
+  become: yes
+  become_user: "{{ devspace_user }}"
+  docker_service:
+    project_src: "{{ devspaceci_home }}"
+    state: absent
+  when: gitstatus.changed
+
+# config
+- name: ensures .ssh dir exists
+  become: yes
+  become_user: "{{ devspace_user }}"
+  file:
+    path: "{{ devspace_home }}/slave/.ssh"
+    state: directory
+    owner: "{{ devspace_user }}"
+    group: "{{ devspace_user }}"
+    mode: 0700
+
+- name: copy snoopy
+  become: yes
+  become_user: "{{ devspace_user }}"
+  copy:
+     src: "{{ snoopy_dir_path }}/{{ item.srcfile }}"
+     dest: "{{ devspace_home }}/slave/{{ item.dest }}"
+     owner: "{{ devspace_user }}"
+     group: "{{ devspace_user }}"
+     mode: 0700
+  with_items:
+    - { srcfile: ".ssh/config", dest: ".ssh/config" }
+    - { srcfile: ".ssh/snoopycrimecop_github", dest: ".ssh/snoopycrimecop_github" }
+    - { srcfile: ".ssh/snoopycrimecop_github.pub", dest: ".ssh/snoopycrimecop_github.pub" }
+    - { srcfile: ".gitconfig", dest: ".gitconfig" }
+  when: snoopy_dir_path|length>0
+
+- name: known_hosts
+  become: yes
+  become_user: "{{ devspace_user }}"
+  shell: ssh-keyscan github.com >> {{ devspace_home }}/slave/.ssh/known_hosts
```

Comment:
Doesn't `ssh-keyscan github.com` always give the same output? Can you hard-code it as a file instead so it's idempotent?